actions/submit-signing-request/utils.ts

```diff
@@ -69,4 +69,48 @@ export function httpErrorResponseToText(err: AxiosError): string {
     }
 
     return err.message;
+}
+
+export function parseUseDefinedParameters(parameters: string): {name:string, value: string}[] {
+    // split value by lines
+    const parmLines = parameters.split('\n');
+    // for each line get param name and value
+    return parmLines.map(parseUseDefinedParameter)
+        .filter(p => p !== null)
+        .map(p => p!);
+}
+
+function parseUseDefinedParameter(line: string): { name: string, value: string } | null {
+    if (!line) {
+        return null;
+    }
+    const nameValueSeparatorIndex = line.indexOf(':');
+    if (nameValueSeparatorIndex === -1) {
+        throw new Error(`Invalid parameter line: ${line}`);
+    }
+    const name = line.substring(0, nameValueSeparatorIndex).trim();
+    const value = line.substring(nameValueSeparatorIndex + 1).trim();
+
+    // validate name
+    if (!name) {
+        throw new Error(`Parameter name cannot be empty. Line: ${line}`);
+    }
+    if(/[a-zA-Z0-9.\-_]+/.exec(name)?.[0] !== name) {
+        throw new Error(`Invalid parameter name: ${name}. Only alphanumeric characters, dots, dashes and underscores are allowed.`);
+    }
+
+    // validate value
+    let parsedValue = null;
+    try{
+        parsedValue = JSON.parse(value);
+    }
+    catch {
+        parsedValue = value;
+    }
+
+    if (`"${parsedValue}"` !== value) {
+        throw new Error(`Invalid parameter value: ${value}. Only valid JSON strings are allowed.`);
+    }
```

Comment:
I would have written:

```
let parsedValue = JSON.parse(value);
if ((typeof parsedValue) !== 'string') throw...
````

Seems shorter

Reply:
Is only two lines instead of 9 and imo easier to read/understand, which just as good.

Reply:
done, the code is more readable, but we still need try/cacth